File: services/assesment-service/src/main/java/sgptt/assessvc/assesmentservice/entity/Academy.java
package sgptt.assessvc.assesmentservice.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.Set;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Entity
public class Academy {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "academy_id")
    private Long academyId;

    @Column(name = "name", nullable = false, length = 50)
    private String name;

    @ManyToMany(fetch = FetchType.LAZY)
    private Set<Protocol> protocols;
}

File: services/protocols-service/src/main/kotlin/com/sgptt/protocolsservice/repository/entity/Academy.kt
package com.sgptt.protocolsservice.repository.entity

import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.FetchType
import jakarta.persistence.GeneratedValue
import jakarta.persistence.GenerationType
import jakarta.persistence.Id
import jakarta.persistence.ManyToMany

@Entity
data class Academy(
	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	@Column(name = "academy_id")
	val academyId: Long,
	
	@Column(name = "name", nullable = false, length = 50)
	val name: String,
	
	@ManyToMany(fetch = FetchType.LAZY)
	val protocols: Set<Protocol>
)